check_pedigree.py

```diff
@@ -0,0 +1,1217 @@
+        if trio not in complete_inferred_trios:
+            given_only_trios.append(trio)
+            out_review.write("given_only_trio\t[s:{trio.s}, pat_id:{trio.pat_id}, mat_id:{trio.mat_id}]\t.\n".format(**locals()))
+
+    trios_accounted = inferred_only_trios + confirmed_trios  #(given only won't be in ht so don't have to account for those)
+
+    accounted_relations = []
+    for trio in trios_accounted:
+        accounted_relations.append(sorted([trio.s, trio.pat_id]))
+        accounted_relations.append(sorted([trio.s, trio.mat_id]))
+        
+   
+    out_summary.write("Number of confirmed trios: " + str(len(confirmed_trios)) + '\n')
+    out_summary.write("Number of inferred only trios: " + str(len(inferred_only_trios)) + '\n')
+    out_summary.write("Number of given only trios: " + str(len(given_only_trios)) + '\n\n')
+
+    # removed the inferred trios that have been accounted for already from the next iteration of the kinship ht
+    ht = ht.filter(hl.literal(accounted_relations).contains(ht.pair), keep=False)
+
+    # annotate master ht with trio annotations
+    master_kin_ht = master_kin_ht.annotate(call = hl.if_else(hl.literal(accounted_relations).contains(master_kin_ht.pair),"parent_offspring", master_kin_ht.call))
+    master_kin_ht = master_kin_ht.annotate(final_evaluation = hl.if_else(hl.literal(inferred_only_trio_pairs).contains(master_kin_ht.pair), "inferred_only_trio_component", master_kin_ht.final_evaluation))
+    master_kin_ht = master_kin_ht.annotate(final_evaluation = hl.if_else(hl.literal(confirmed_trio_pairs).contains(master_kin_ht.pair), "confirmed_trio_component", master_kin_ht.final_evaluation))
+
+    return(ht, master_kin_ht, given_only_trios, complete_inferred_trios, inferred_only_trios)
+
+
+
+def evaluate_duos(
+    ht: hl.Table,
+    master_kin_ht: hl.Table,
+    out_summary: io.TextIOWrapper,
+    out_review: io.TextIOWrapper,
+    ped_duos: list
+) -> Tuple[hl.Table, hl.Table, list, list]:
+
+    """
+    Determine if duos are confirmed, inferred only, or given only, annotate master kinship table with duo decisions
+    :param ht: hl.Table
+    :param master_ht: hl.Table
+    :return: ht with duos removed and master kinship ht with annotated duos decisions, inferred duos, and given-only duos
+    :rtype: Tuple[hl.Table, hl.Table, list, list]
+    """
+    
+    parent_child_ht = ht.filter(ht.relationship == PARENT_CHILD)
+    parent_child_ht_i = parent_child_ht.aggregate(hl.agg.group_by(parent_child_ht.i, hl.agg.collect(parent_child_ht.j)))
+    parent_child_ht_j = parent_child_ht.aggregate(hl.agg.group_by(parent_child_ht.j, hl.agg.collect(parent_child_ht.i)))
+
+    # for at least one of the samples in the parent-offspring pair, that sample must appear in a parent offspring pair only once.
+    potential_duos = {**parent_child_ht_i , **parent_child_ht_j}  # combine the two dictionaries
+
+
+    duos = []
+    for sample,possible_parents in potential_duos.items():
```

Comment:
Since the `i` and `j` aren't sorted, I don't think we should use `sample` and `possible_parents` here -- it really is just `sample1` and `sample2`, or `duo_s1`, or something.